accesspoc/dips/views.py

```diff
@@ -427,26 +441,36 @@ def new_dip(request):
     if request.method == 'POST' and dip_form.is_valid() and dc_form.is_valid():
         dip = dip_form.save(commit=False)
         dip.dc = dc_form.save()
+        # TODO: Avoid this save from updating the related ES
+        # documents, as a later save when the async_result id
+        # is added, will repeat the process. It may require
+        # stop using signals and move to a custom save method.
+        dip.save()
+
+        # Extract and parse METS file asynchronously
+        async_result = extract_and_parse_mets.delay(dip.pk, dip.objectszip.path)
+        # Save the async_result id to relate later with the TaskResult
+        # related object, which is not created on task call. Celery docs
+        # recommend to call get() or forget() on AsyncResult to free the
+        # backend resources used to store and transmit results, but the
+        # former waits for the task completion and the later may remove
+        # the result in the backend and the database. The only solution
+        # I could find is to rely on the `__del__` method, which cancels
+        # pending operations over the AsyncResult but doesn't delete the
+        # related TaskResult from the database.
+        dip.import_task_id = async_result.id
+        dip.import_status = DIP.IMPORT_PENDING
         dip.save()
 
-        # Extract METS file from DIP objects zip
-        tmpdir = tempfile.mkdtemp()
-        if not os.path.isdir(tmpdir):
-            os.mkdirs(tmpdir)
-        objectszip = os.path.join(settings.MEDIA_ROOT, request.FILES['objectszip'].name)
-        metsfile = ''
-        zip = zipfile.ZipFile(objectszip)
-        for info in zip.infolist():
-            if re.match(r'.*METS.[0-9a-f\-]{32}.*$', info.filename):
-                print('METS file to extract:', info.filename)
-                metsfile = zip.extract(info, tmpdir)
-        # Parse METS file
-        mets = METS(os.path.abspath(metsfile), dip.pk)
-        mets.parse_mets()
-        # Delete extracted METS file
-        shutil.rmtree(tmpdir)
+        # Show notification to user about import in progress
+        messages.info(request, _(' '.join([
+            'A background process has been launched to extract and parse',
+            'the METS file. After the process finishes and the interface',
+            'is reloaded, a link to the Folder will show up in the',
+            'Folders table at the related Collection page.',
+        ])))
```

Comment:
Is that going to work? I would have thought that the join would never happen because the annotated string is extracted beforehand? An alternative could be:

```python
        messages.info(
            request,
            'A background process has been launched to extract and parse the'
            ' METS file. After the process finishes and the interface is'
            ' reloaded, a link to the Folder will show up in the Folders table'
            ' at the related Collection page.')
```

Reply:
Oh, forgot to change it here from the previous feedback.